net_transport_test.go

```diff
@@ -84,16 +84,16 @@ func TestNetworkTransport_CloseStreams(t *testing.T) {
 		wg := &sync.WaitGroup{}
 		wg.Add(5)
 
+		var errCounter int32
 		appendFunc := func() {
 			defer wg.Done()
 			var out AppendEntriesResponse
 			if err := trans2.AppendEntries("id1", trans1.LocalAddr(), &args, &out); err != nil {
-				t.Fatalf("err: %v", err)
-			}
-
-			// Verify the response
-			if !reflect.DeepEqual(resp, out) {
-				t.Fatalf("command mismatch: %#v %#v", resp, out)
+				atomic.AddInt32(&errCounter, 1)
+				t.Errorf("err: %v", err)
+			} else if !reflect.DeepEqual(resp, out) { // Verify the response
+				atomic.AddInt32(&errCounter, 1)
+				t.Errorf("command mismatch: %#v %#v", resp, out)
```

Comment:
Thank you for cleaning up the calls to `t.Fatal` in a goroutine! 

While this is good cleanup, I think it would be better to do this cleanup in a separate PR, and at the very least in a separate commit. Keeping commits isolated to one change makes the git history much easier to read, and can make it easier to cherry-pick changes.

I'd also like to suggest a different approach to fixing this problem. Instead of an atomic counter, how about returning an error, and using https://pkg.go.dev/golang.org/x/sync/errgroup to run the goroutines. That is a pattern I've used before, and it works really well. That might look something like this:


```go
	appendFunc := func() error {
		// return errors instead of t.Fatalf
	}

	g := new(errgroup.Group)
	for i = 0; i < 5; i++ {
		g.Go(appendFunc())
	}
	require.NoError(t, g.Wait())
```

This will remove the need for a `sync.WaitGroup` as well.

Reply:
That's great idea to use `ErrGroup`. I would make the fix in another PR as your suggestion.